src/core_ghidra.cpp

```diff
@@ -132,6 +133,87 @@ static void ApplyPrintCConfig(RConfig *cfg, PrintC *print_c)
 	print_c->setMaxLineSize(cfg_var_linelen.GetInt(cfg));
 }
 
+RAnnotatedCode* r2ghidra_decompile_annotated_code(RCore *core){
```

Comment:
I would prefer if the internal implementation function passed offset explicitly instead of relying on global offset value. If necessary the core->offset can be passed as default position but that should be done at user interface level in the corresponding r2 command handler.

Reply:
> I would prefer if the internal implementation function passed offset explicitly instead of relying on global offset value.

What benefit do we get from this? Is `static void Decompile(RCore *core, DecompileMode mode)` also not doing the same thing?

Reply:
Yes there are other parts of code doing this. Poor separation of user interface from the implementation of main logic isn't uncommon in r2 and it makes it's API more difficult to use.

There are two aspects to this.

 One is that "global variables  are evil".  `core->offset`  isn't a one from language perspective but from the perspective of data-flow in the whole system it is one, with the same problems. You can find plenty of articles on this topic.

Other aspect is that core->offset is mixing responsibilities of 3 different concepts.
* core->offset as part of interface  - this only makes sense in the r2 command mode, in Cutter and Panel mode there is no single current offset.
* current offset as implementation detail of some io backends
* function capable of decompiling code at any point of program

Decompilation function shouldn't know anything about the current interface state. It is the responsibility of user interface tracking which parts you are looking at and to request decompiling them.

Reply:
I am joining @karliss here with these concerns. 
@NirmalManoj please suggest better API layer functions to demonstrate the desired changes, show us and then implement if it looks good :)

Reply:
Can we change `RAnnotatedCode* r2ghidra_decompile_annotated_code(RCore *core)` to `RAnnotatedCode* r2ghidra_decompile_annotated_code(RCore *core, RVA addr)`? 

With this change,  in `r2ghidra_decompile_annotated_code` we decompile code at the address `RVA addr` without using core->offset.  @karliss @ITAYC0HEN

Reply:
I have made this change in the new commit @karliss @ITAYC0HEN . Now, cutter's plugin doesn't use core->offset directly or indirectly(as far as I could see:))

Reply:
Looks good.

Reply:
@thestr4ng3r Should I make similar changes for retdec in the current PR?

Reply:
Yes please.

Reply:
@thestr4ng3r Done. Please check if I have missed someplace that fetches offset from the core.